x-pack/test/api_integration/apis/ml/data_visualizer/get_field_stats.ts

```diff
@@ -0,0 +1,238 @@
+            { key: 989.763916015625, doc_count: 2 },
+            { key: 991.290771484375, doc_count: 2 },
+            { key: 992.0765991210938, doc_count: 2 },
+            { key: 993.8115844726562, doc_count: 2 },
+            { key: 993.8973999023438, doc_count: 2 },
+            { key: 994.0230102539062, doc_count: 2 },
+            { key: 994.364990234375, doc_count: 2 },
+            { key: 994.916015625, doc_count: 2 },
+          ],
+          topValuesSampleSize: 4260,
+          topValuesSamplerShardSize: -1,
+        },
+      ],
+    },
+  };
+
+  const nonMetricFieldsTestData = {
+    testTitle: 'returns stats for non-metric fields specifying query and time range',
+    index: 'ft_farequote',
+    user: USER.ML_POWERUSER,
+    requestBody: {
+      query: {
+        bool: {
+          must: {
+            term: { airline: 'AAL' },
+          },
+        },
+      },
+      fields: [
+        { fieldName: '@timestamp', type: 'date', cardinality: 4751 },
+        { fieldName: '@version.keyword', type: 'keyword', cardinality: 1 },
+        { fieldName: 'airline', type: 'keyword', cardinality: 19 },
+        { fieldName: 'type', type: 'text', cardinality: 0 },
+        { fieldName: 'type.keyword', type: 'keyword', cardinality: 1 },
+      ],
+      samplerShardSize: -1, // No sampling, as otherwise counts would vary on each run.
+      timeFieldName: '@timestamp',
+      earliest: 1454889600000, // February 8, 2016 12:00:00 AM GMT
+      latest: 1454976000000, // February 9, 2016 12:00:00 AM GMT
+      maxExamples: 10,
+    },
+    expected: {
+      responseCode: 200,
+      responseBody: [
+        { fieldName: '@timestamp', count: 1733, earliest: 1454889602000, latest: 1454975948000 },
+        {
+          fieldName: '@version.keyword',
+          isTopValuesSampled: false,
+          topValues: [{ key: '1', doc_count: 1733 }],
+          topValuesSampleSize: 1733,
+          topValuesSamplerShardSize: -1,
+        },
+        {
+          fieldName: 'airline',
+          isTopValuesSampled: false,
+          topValues: [{ key: 'AAL', doc_count: 1733 }],
+          topValuesSampleSize: 1733,
+          topValuesSamplerShardSize: -1,
+        },
+        {
+          fieldName: 'type.keyword',
+          isTopValuesSampled: false,
+          topValues: [{ key: 'farequote', doc_count: 1733 }],
+          topValuesSampleSize: 1733,
+          topValuesSamplerShardSize: -1,
+        },
+        { fieldName: 'type', examples: ['farequote'] },
+      ],
+    },
+  };
+
+  const errorTestData = {
+    testTitle: 'returns error for index which does not exist',
+    index: 'ft_farequote_not_exists',
+    user: USER.ML_POWERUSER,
+    requestBody: {
+      query: { bool: { must: [{ match_all: {} }] } },
+      fields: [
+        { type: 'number', cardinality: 0 },
+        { fieldName: 'responsetime', type: 'number', cardinality: 4249 },
+      ],
+      samplerShardSize: 1000, // No sampling, as otherwise counts could vary on each run.
```

Comment:
This value is set to `-1` for the other test data with no sampling. Is the `1000` intended?

Reply:
I've set it to `-1` in https://github.com/elastic/kibana/pull/64165/commits/2b63f4504c03ec5d78ccef7cba518a1511be3b15 to be consistent with the other tests, although as this test should fail as it's with the unauthorized user, the `samplerShardSize` should be irrelevant.